File: src/main/java/org/distril/beengine/player/data/PlayerData.java
package org.distril.beengine.player.data;

import com.nukkitx.math.vector.Vector3f;
import lombok.Getter;
import lombok.Setter;
import org.distril.beengine.server.Server;
import org.distril.beengine.world.util.Location;

@Setter
@Getter
public class PlayerData {

	private float pitch, yaw, headYaw;
	private Location location = Location.from(Vector3f.from(0, 60, 0),
			Server.getInstance().getWorldRegistry().getDefaultWorld());
	private GameMode gameMode = GameMode.CREATIVE;
}

File: src/main/kotlin/org/distril/beengine/player/data/PlayerData.kt
package org.distril.beengine.player.data

import com.nukkitx.math.vector.Vector3f
import org.distril.beengine.server.Server
import org.distril.beengine.world.util.Location

class PlayerData {

    var pitch = 0f
    var yaw = 0f
    var headYaw = 0f
    var location = Location(
        Server.worldRegistry.defaultWorld,
        Vector3f.from(0f, 60f, 0f)
    )
    var gameMode = GameMode.CREATIVE
}
